codecarbon/core/rapl.py

```diff
@@ -0,0 +1,29 @@
+from dataclasses import dataclass
+
+from codecarbon.core.units import Energy, Time
+
+
+@dataclass
+class RAPLFile:
+    name: str
+    path: str
+    energy_reading: float
+    power_measurement: float
+
+    def _get_value(self) -> float:
+        """
+        Reads the value in the file at the path
+        """
+        with open(self.path, "r") as f:
+            return Energy.from_ujoules(float(f.read()))
+
+    def start(self):
+        self.energy_reading = self._get_value()
+        return
+
+    def end(self, delay):
+        self.power_measurement = (
+            abs(self._get_value() - self.energy_reading)
```

Comment:
should we rather take average here of the last reading and current reading ?
What if the two Energy readings are same, it will output 0.
@vict0rsch @Lothiraldan @sorelle

Reply:
We chatted about that and concluded this is fine, resolving